viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    if not title or not genre or not rating: 
+        return None
+    
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+
+    if not user_data or not movie: 
+        return None
+    
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    if not user_data or not movie: 
+        return None
+    
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+
+            user_data["watchlist"].remove(movie)
+
+            user_data["watched"].append(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    
+    total_rating = 0.0
+    avg_rating = 0.0
+
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"]
+
+    avg_rating = total_rating/len(user_data["watched"])

```

Comment:
We will want to follow the Python convention of surrounding operators like `/` with white space.
```suggestion
    avg_rating = total_rating / len(user_data["watched"])

```